### PR TITLE
Bump Xcode to 11.2.1

### PR DESCRIPTION
**Pull request by bayandin**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
Xcode 11.2 has been released https://developer.apple.com/documentation/xcode_release_notes/xcode_11_2_release_notes

### CODE REVIEW

**Comment by bayandin** [comment]
Thanks!
Actually, maybe it will be wise to wait for 11.2.1 release, looks like it's going to happen soon, GM already published: https://developer.apple.com/documentation/xcode_release_notes/xcode_11_2_1_gm_seed_release_notes/

**Comment by bayandin** [comment]
Hey @MikeMcQuaid, 

Xcode 11.2.1 has been released and I'm going to bump the version in this PR, but I'm not quite sure what to do with `detect_version_from_clang_version`: Last time I've just added a new entry
```diff
         case (DevelopmentTools.clang_version.to_f * 10).to_i
         when 0       then "dunno"
         when 60      then "6.0"
         when 61      then "6.1"
         when 70      then "7.0"
         when 73      then "7.3"
         when 80      then "8.0"
         when 81      then "8.3"
         when 90      then "9.2"
         when 91      then "9.4"
         when 100     then "10.2.1"
         when 110     then "11.0"
         when 111     then "11.1"
-        else              "11.1"
+        when 112     then "11.2"
+        else              "11.2"
         end
```
without really checking what `(DevelopmentTools.clang_version.to_f * 10).to_i` returns (for Xcode 11.2).

Now I've updated Xcode to 11.2.1 locally and `(DevelopmentTools.clang_version.to_f * 10).to_i` returns `110`. And it seems for Xcode 11.1 it returns `110` (insted of `111`) as well.

**Comment by gfguthrie** [comment]
This might've been where things have gotten off the rails – perhaps it should just be `when 110 then 11.2.1` – all the other 11s should go away. It just happened to be that the version coincided with initial 11.0 release, but Apple isn't bumping the clang version with every OS release.

**Comment by bayandin** [comment]
> This might've been where things have gotten off the rails – perhaps it should just be `when 110 then 11.2.1` – all the other 11s should go away. It just happened to be that the version coincided with initial 11.0 release, but Apple isn't bumping the clang version with every OS release.

Yep, my thoughts were the same.

Force pushed 11.2.1 version

File: Library/Homebrew/os/mac/xcode.rb [hunk 1]
```diff
@@ -176,9 +176,8 @@ def detect_version_from_clang_version
         when 90      then "9.2"
         when 91      then "9.4"
         when 100     then "10.2.1"
-        when 110     then "11.0"
-        when 111     then "11.1"
-        else              "11.1"
+        when 110     then "11.2"
+        else              "11.2"
```
Comment (gfguthrie) [inline Library/Homebrew/os/mac/xcode.rb]: should these also be `11.2.1`?

Comment (bayandin) [reply Library/Homebrew/os/mac/xcode.rb]: Whoops 😬
Thanks, fixed!

**Comment by MikeMcQuaid** [comment]
@bayandin Need to wait for our CI nodes to be updated for this. I expect to do that next week. We've been waiting on a matching CLT version.

**Comment by bayandin** [comment]
Thanks!

**Comment by MikeMcQuaid** [comment]
CI upgraded. 
Thanks so much for your contribution! Without people like you submitting PRs we couldn't run this project. You rock, @bayandin!

**Comment by bayandin** [comment]
Amazing! Thank you, Mike!

**Pull request merged**